cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -986,7 +986,9 @@ std::vector<timestamp> LocalVersionedEngine::batch_get_update_times(
     return results;
 }
 
-std::shared_ptr<std::unordered_map<std::pair<StreamId, VersionId>, AtomKey>> LocalVersionedEngine::get_stream_index_map(
+std::pair<std::shared_ptr<std::unordered_map<std::pair<StreamId, VersionId>, AtomKey>>, 
```

Comment:
This type is very unwieldy, at the least it would benefit from some type aliases to make it more readable

Reply:
done

Reply:
I want to refactor this whole method, doing two calls to batch_read_latest then batch_read_specific is a bit nuts. What we need is a method in version_store_batch_methods.hpp that takes a list of StreamId and VersionQuery, and returns a vector of optional<AtomKey>. That way we can preserve the order (which is lost at the moment because of the two calls to get the keys depending on the latest vs specific, which removes the need for an unordered_map.